### ISSUE TITLE
We need a better changelog / how to update, especially on breaking changes

### ISSUE THREAD

**Issue by jimtng** [boc-tothefuture/openhab-jruby#216]
Unless someone has been involved in the actual breaking change process, it isn't immediately apparent for example that Group.items should be changed to Group.members in rules, and that Group.group no longer refers to the group item and it should be changed to just Group.

For a casual user of the library, changes like these that directly affect their rules should be highlighted and documented in an easy to understand manner, without requiring them to analyse the difference in docs between versions.

**Comment by boc-tothefuture** [comment]
It is really up to us what goes into the changelog as it is part of the commit message.  It can be as long and thorough as we want.

**Comment by jimtng** [comment]
Yeah I (we) tend to make my (our) commit messages as short and brief as possible, bordering cryptic. I'll keep it in mind for the future to write a more descriptive message.

**Comment by jimtng** [comment]
Can / should we manually update the changelog for this 3.0? It may not matter since we don't seem to have many users yet.

**Comment by pacive** [comment]
If you have specified the gem version in jruby.cfg as ~>2.0 it doesn't get updated automatically right? That should prevent some of these issues for users at least. But should also be noted somewhere that they need to change this to receive further updates.

**Comment by jimtng** [comment]
> If you have specified the gem version in jruby.cfg as ~>2.0 it doesn't get updated automatically right? That should prevent some of these issues for users at least. But should also be noted somewhere that they need to change this to receive further updates.

Correct. However, sooner or later they'd want to / need to upgrade, and when that time comes, they'd need to know "so what do I need to change in my code once I've upgraded to this version?", and that's the information that we need to provide in the changelog too.

**Comment by pacive** [comment]
Agreed, this needs to be well documented. Should we add a page to the docs that outlines the upgrade steps in the meantime? And make sure to have better commit messages in the future.

**Comment by boc-tothefuture** [comment]
On this one we have two actions:
1. ensure better commit messages
2. Better upgrade docs

@pacive or @jimtng do either of you want to tackle the docs, attach to this issue so we can close this one out?

**Comment by jimtng** [comment]
Are there any specific issues that need to be mentioned in the current docs? I was referring about the changelog in this issue and now that I am aware that commit messages can be as long as needed, I believe this issue can be closed.

**Comment by boc-tothefuture** [comment]
I think the issue was around upgrading when versions change, specifically breaking version changes 2 -> 3, etc.
